contracts/interfaces/IAssessment.sol

```diff
@@ -0,0 +1,175 @@
+   *  Holds data for a vote belonging to an assessor.
+   *
+   *  The structure is used to keep track of user's votes. Each vote is used to determine
+   *  a user's share of rewards or to create a fraud resolution which excludes fraudulent votes
+   *  from the initial poll.
+   */
+  struct Vote {
+   // Identifier of the claim or incident
+    uint104 eventId;
+   // If the assessor votes to accept the event it's true otherwise it's false
+    bool accepted;
+   // Date and time when the vote was cast
+    uint32 timestamp;
+   // How many tokens were staked when the vote was cast
+    uint96 tokenWeight;
+   // Can be a claim or an incident (See EventType enum)
+    uint8 eventType;
+  }
+
+  struct Poll {
+    uint96 accepted;
+    uint96 denied;
+    uint32 start;
+    uint32 end;
+  }
+
+  /*
+   *  Holds the requested amount, NXM price, submission fee and other relevant details
+   *  such as parts of the corresponding cover details and the payout status.
+   *
+   *  This structure has snapshots of claim-time states that are considered moving targets
+   *  but also parts of cover details that reduce the need of external calls. Everything is fitted
+   *  in a single word that contains:
+   */
+  struct ClaimDetails {
+   // Amount requested as part of this claim up to the total cover amount
+    uint96 amount;
+   // The identifier of the cover on which this claim is submitted
+    uint32 coverId;
+   // Cover period represented as days, used to calculate rewards
+    uint16 coverPeriod;
+   // The asset which is expected at payout. E.g ETH, DAI (See Asset enum)
+    uint8 payoutAsset;
+   // The price (TWAP) of 1 NXM in the covered asset, at claim-time
+    uint80 nxmPriceSnapshot;
+   // A snapshot of CLAIM_ASSESSMENT_DEPOSIT_PERC if it is changed before the payout
+    uint16 assessmentDepositPerc;
+   // True when the payout is redeemed. Prevents further payouts on the claim.
+    bool payoutRedeemed;
+  }
+
+  struct Claim {
+    Poll poll;
+    ClaimDetails details;
+  }
+
+  /*
+   *  Keeps details related to incidents.
+   */
+  struct IncidentDetails {
+    // Product identifier
+    uint24 productId;
+    // Timestamp marking the date of the incident used to verify the user's eligibility for a claim
+    // according to their cover period.
+    uint32 date;
+    // The asset which is expected at payout. E.g ETH, DAI (See Asset enum).
+    uint8 payoutAsset;
+    // A snapshot of INCIDENT_IMPACT_ESTIMATE_PERC if it changes while voting is still open.
+    uint96 activeCoverAmount;
+    // A copy of INCIDENT_IMPACT_ESTIMATE_PERC if it changes while voting is still open.
+    uint16 impactEstimatePerc;
+    // A copy of INCIDENT_ASSESSMENT_DEPOSIT_PERC if it changes while voting is still open.
+    uint16 assessmentDepositPerc;
+    // True when the assessment deposit has already been redeemed and false otherwise.
+    bool depositRedeemed;
+  }
+
+  struct AffectedToken {
+    uint96 priceBefore;
+    address contractAddress;
```

Comment:
needs `productId`, not `contractAddress`

Reply:
Needs the contract address of the ERC token, not the product id which is a uint24. This is needed to make the actual transfer when redeeming yield token cover payouts.

Reply:
The point here is that you'll need to check if the token address is the same as the product token otherwise you could pass WBTC instead of DAI for example (by mistake or malice).

Reply:
That depends on the product. It will be copied from the product when the incident is submitted. This struct actually prevents "mistake or malice" by relying on external data at all times instead of once. In other words it is a snapshot that makes things both safer and cheaper.